pkg/system/prlimit.go

```diff
@@ -68,19 +69,58 @@ func SetCPUTimeLimit(pid int, value uint64) error {
 	return limiter.SetCPUTimeLimit(pid, value)
 }
 
+// QEMU progress report uses carriage returns for the percentages, the default split function
+// in Scanner doesn't handle carriage returns.
+func ScanLinesWithCR(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
does this need to b exported? At least for now it's only called in this package bu the export ExecWithLimits func.

Reply:
No it doesn't this is specifically because QEMU-IMG reports the progress with carriage returns instead of new lines, so I had to alter the split function to work with them.